packages/amazon-pay-integration/src/amazon-pay-v2-button-strategy.ts

```diff
@@ -0,0 +1,145 @@
+    CheckoutButtonStrategy,
+    getShippableItemsCount,
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType,
+    PaymentIntegrationService,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import {
+    isWithBuyNowFeatures,
+    WithAmazonPayV2ButtonInitializeOptions,
+    WithBuyNowFeature,
+} from './amazon-pay-v2-button-options';
+import AmazonPayV2RequestSender from './amazon-pay-v2-request-sender';
+import AmazonPayV2ConfigCreationError from './errors/amazon-pay-v2-config-creation-error';
+
+export default class AmazonPayV2ButtonStrategy implements CheckoutButtonStrategy {
+    private _buyNowInitializeOptions: WithBuyNowFeature['buyNowInitializeOptions'];
+
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private amazonPayV2PaymentProcessor: AmazonPayV2PaymentProcessor,
+        private amazonPayV2ConfigRequestSender: AmazonPayV2RequestSender,
+    ) {}
+
+    async initialize(
+        options: CheckoutButtonInitializeOptions & WithAmazonPayV2ButtonInitializeOptions,
+    ): Promise<void> {
+        const { methodId, containerId, amazonpay } = options;
+        const { buttonColor } = amazonpay || {};
+
+        if (!methodId || !containerId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "methodId" or "containerId" argument is not provided.',
+            );
+        }
+
+        const { getPaymentMethodOrThrow } = this.paymentIntegrationService.getState();
+
+        const paymentMethod = getPaymentMethodOrThrow<AmazonPayV2InitializeOptions>(methodId);
+        const { initializationData } = paymentMethod;
+
+        await this.amazonPayV2PaymentProcessor.initialize(paymentMethod);
+
+        if (!amazonpay) {
+            await this.paymentIntegrationService.loadDefaultCheckout();
+        }
+
+        const initializeAmazonButtonOptions = isWithBuyNowFeatures(amazonpay)
+            ? undefined
+            : amazonpay;
+
+        if (
+            isWithBuyNowFeatures(amazonpay) &&
+            typeof amazonpay.buyNowInitializeOptions?.getBuyNowCartRequestBody === 'function'
+        ) {
+            this._buyNowInitializeOptions = amazonpay.buyNowInitializeOptions;
+            this.amazonPayV2PaymentProcessor.updateBuyNowFlowFlag(true);
+        }
+
+        this.amazonPayV2PaymentProcessor.renderAmazonPayButton({
+            checkoutState: this.paymentIntegrationService.getState(),
+            containerId,
+            methodId,
+            options: initializeAmazonButtonOptions,
+            placement: AmazonPayV2Placement.Cart,
+            buttonColor,
+            isButtonMicroTextDisabled: initializationData?.isButtonMicroTextDisabled,
+        });
+
+        if (this._buyNowInitializeOptions) {
+            this.amazonPayV2PaymentProcessor.prepareCheckoutWithCreationRequestConfig(
+                this._getCheckoutCreationRequestConfig.bind(this),
+            );
+        }
+    }
```

Comment:
Don't you want to separate initialization for buyNow and non-buyNow flows within different methods?

Reply:
It might help to simplify and improve code readability